### ISSUE TITLE
Performance is really slow

### ISSUE THREAD

**Issue by tintin10q** [valentjn/vscode-ltex#611]
**Describe the bug**
It takes about 10 seconds to check my file. Then after I fix one error I have to wait 10 seconds again. The fact that checking the file takes about 10 seconds is acceptable of course but in my opinion waiting 10 seconds after each correction is not. 

So I can click on a problem to show the fixes and then on another problem to show the fixes but if I actually apply a fix all the fixes are gone and I have to wait another 10 or so seconds before I can apply another fix. Is this just me or is that normal?

So just to be clear using this plugin I get the option to fix one problem while I can see the fixes for all the problems but if I actually apply a fix I have to wait again until I can apply another fix. 

I also changed the settings to only recheck the document on save. I also tried to check manually but the behavior is the same. 

**Steps to reproduce**

1. Create a document with some mistakes in it
2. Wait for the plugin to do its thing or activate it manually
3. Observe a couple quick fixes
4. Apply a quick fix
5. Observer that now the quick fixes for the other mistakes are gone 
6. Observer that you now have to wait until the whole document is checked again until you can fix another problem

**Expected behavior**
I would expect that I can just keep fixing mistakes after one check and that I don't have to wait again after applying a fix for a mistake. 

For instance if I paste text in a browser I can keep clicking the spelling errors and don't have to wait after applying a fix. 

**Sample document**
At first I thought it would only happen with a large document but I tried it with a small document and the same happens. 

<details>
I have also tried to assign more ram to the language model (about 5 gb) and 5 gb to the cache but this did not do anything.

</details>

**LTeX configuration**
Please paste all configuration settings starting with `ltex.` from your `settings.json`. You can help us by temporarily removing some irrelevant settings from your `settings.json` and see if the bug still occurs.

<details>

```
{
    "telemetry.enableTelemetry": false,
    "telemetry.enableCrashReporter": false,
    "workbench.editorAssociations": {
        "*.ipynb": "jupyter.notebook.ipynb"
    },
    "redhat.telemetry.enabled": false,
    "languageToolLinter.languageTool.language": "en-GB",
    "languageToolLinter.languageTool.ignoredWordsGlobal": [ ],
    "window.zoomLevel": 1,
    "editor.minimap.enabled": false,
    "workbench.colorTheme": "Default High Contrast",
    "cSpell.language": "en-GB",
    "ltex.additionalRules.enablePickyRules": true,
    "ltex.additionalRules.languageModel": "C:\\Users\\tintin10q\\onpath\\languagetool\\languagemodels",
    "ltex.additionalRules.motherTongue": "nl",
    "ltex.additionalRules.neuralNetworkModel": "C:\\Users\\tintin10q\\onpath\\languagetool\\word2vec",
    "ltex.additionalRules.word2VecModel": "C:\\Users\\tintin10q\\onpath\\languagetool\\word2vec",
    "ltex.checkFrequency": "manual",
    "ltex.completionEnabled": true,
    "ltex.disabledRules": {
        
    
    },
    "ltex.java.maximumHeapSize": 5000,
    "ltex.language": "en-GB",
    "ltex.sentenceCacheSize": 5000,
    "ltex.statusBarItem": true
}
```

</details>

**Version information**
List here the version information of the relevant software.

- Operating system Windows 10
- VS Code: 1.67.1
- vscode-ltex: x.xx

**Comment by tintin10q** [comment]
I think the problem is that it will recheck the file after each fix but that makes everything really slow.

**Comment by solarchemist** [comment]
I agree with @tintin10q. Everything works, but every operation takes far too long (many seconds before the "Quick fix..." link even becomes visible when hovering over an underlined word, and this occurs again for every word even with `ltex.checkFrequency=manual`).

Working through a single R Sweave document of around 4000 lines is going to take forever.
Using keyboard shortcuts made no difference.

I am a first-time user of this extension, but to me it appears as if LTeX is still running a full check before *every* user operation (it keeps saying `LTeX is checking...` in the statusbar almost every time I hover over an underlined word).

I wonder if this slowness is commonly observed? No other issues mention slowness, which suggests the problem might be caused by something in my configuration? My [VSCodium configuration is set by this Ansible role](https://codeberg.org/ansible/editor-vscodium) (apart from my LTeX config, see below).

```
solarchemist:~/.config/VSCodium/User
$ grep ltex settings.json 
   "ltex.language": "en-GB",
   "ltex.checkFrequency": "manual"
```

`vscode-ltex v13.1.0` on VSCodium:
```
Version: 1.82.2
Release: 23257
Commit: a5004432af6600c99fa4f918a3f2f8df44df71e6
Date: 2023-09-14T18:27:11.315Z
Electron: 25.8.1
ElectronBuildId: undefined
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 6.2.0-34-generic
```
